src/traits.rs

```diff
@@ -58,6 +58,67 @@ pub trait Context {
         hostcalls::enqueue_shared_queue(queue_id, value)
     }
 
+    /// Sends HTTP request with serialized `headers`, `body`, and serialized trailers to upstream.
+    ///
+    /// `on_http_call_response()` will be called with `token_id` when the response is received by the host, or after the timeout.
+    ///
+    /// # Arguments
+    ///
+    /// * `upstream` - The name of the upstream to send the request to (in your envoy configuration)
+    /// * `headers` - The headers to send with the request (in the form of `Vec<(&str, &str)>`)
+    /// * `body` - The body of the request (in the form of `Option<&[u8]>` - `None` if no body)
+    /// * `trailers` - The trailers to send with the request (in the form of `Vec<(&str, &str)>`)
+    /// * `timeout` - The timeout for the request
+    ///
+    /// # Returns
+    ///
+    /// * `OK` on success.
+    /// * `BAD_ARGUMENT` for unknown upstream, or when headers are missing required `:authority`, `:method` and/or `:path` values.
+    /// * `INTERNAL_FAILURE' when the host failed to send requested HTTP call.
+    /// * `INVALID_MEMORY_ACCESS` when `upstream_data`, `upstream_size`, `headers_data`, `headers_size`, `body_data`, `body_size`, `trailers_data`, `trailers_size` and/or `return_call_id` point to invalid memory address.
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// use proxy_wasm::traits::*;
+    /// use proxy_wasm::types::*;
+    /// use std::time::Duration;
+    /// use log::warn;
```

Comment:
I don't think that those examples are verified/build as part of the CI, since you're importing `warn` here, but using `info` in the code.

Reply:
I found [this](https://doc.rust-lang.org/rustdoc/write-documentation/documentation-tests.html). But i am not sure how i can test these examples without running envoy (in a container) and building/loading the examples into a wasm-plugin...

Reply:
It's not about testing examples, but about making sure that the code in documentation even compiles.

Reply:
@antonengelhardt it looks that you can use the `no_run` attribute to skip running of the code as part of the documentation testing:
```
/// ```no_run
/// ...code...
/// ```
```

Reply:
Thanks for the hint! I am a bit busy with work and other things, but i will continue with these tests when i have time :)

Reply:
@PiotrSikora The docs should compile now when running with `cargo test --doc`